examples/rh-sso/README.md

```diff
@@ -0,0 +1,112 @@
+- API - You can use the 3scale [echo API](https://echo-api.3scale.net:443)
+- Client Application to consume the API - You can use a client such as [Postman](https://www.getpostman.com/)
+- APIcast instance
+
+## Red Hat Single Sign-On Configuration for APIcast
+
+1. [Create a new realm](https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html/getting_started_guide/create_a_realm_and_user#create-realm) (different from Master)
+2. Set up Tokens Policies (Realm Settings > Tokens) - e.g to configure access token ttl
+3. Set up [Initial Access Tokens](https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html/securing_applications_and_services_guide/client_registration). This is necessary to synchronise client registrations between 3scale and Red Hat Single Sign-On.
+    - Define Expiration - This will determine how long the access token used to register clients from APIcas will be valid for, so make sure to choose a long lived value unless you want to be changing this token often.
+    - Define Count - This will determine how many clients can be registered using this access token, make sure to choose a large value unless you want to be changing this token often.
+4. [Create some test End-Users](https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.0/html/getting_started_guide/create_a_realm_and_user#create-new-user)
+
+## Client Registration in APIcast (Optional)
+
+In order to authenticate clients in RH SSO and correctly track usage in 3scale, client credentials need to be synchronised between the 2 systems. Typically this functionality would fall outside the realm of the Gateway, but we will show how you can implement this within the Gateway to provide a self contained environment.
+
+The following instructions are only required if you want to manage client registrations from APIcast. Otherwise you can skip this whole section.
+
+### Pre-requisites
+
+- Redis server needs to be running
+
+### APIcast Set up
+
+APIcast v2 should be installed as usual with the following additional dependencies:
+
+- The following dependencies need to be installed via yum
+    - expat-devel: `sudo yum install expat-devel`
+    - C compiler and development tools: `sudo yum group install "Development Tools"`
+- The following lua libraries should be installed using luarocks: 
+    - luaexpat: `sudo luarocks install luaexpat --tree=/usr/local/openresty/luajit`
+    - luaxpath: `sudo luarocks install luaxpath --tree=/usr/local/openresty/luajit`
+    - cjson: `sudo luarocks install lua-cjson --tree=/usr/local/openresty/luajit`
+- And a symbolic link created: so openresty can find them
+    `ln -s /usr/local/openresty/luajit/lib64/lua/5.1/* /usr/local/openresty/luajit/lib/lua/5.1/` 
```

Comment:
When openresty can't find them it will print nice error with all the load paths where they can be.
So it should be installed to one of them that make the most sense.